rule/exported.go

```diff
@@ -174,12 +174,58 @@ func (w *lintExported) lintFuncDoc(fn *ast.FuncDecl) {
 		return
 	}
 
-	prefix := fn.Name.Name + " "
-	if !strings.HasPrefix(firstCommentLine, prefix) {
+	expectedPrefix := fn.Name.Name + " "
+	if strings.HasPrefix(firstCommentLine, expectedPrefix) {
+		return
+	}
+
+	if !w.HasPrefixFold(firstCommentLine, expectedPrefix) {
+		// The comment does not start with the expected prefix.
 		w.addFailuref(fn.Doc, 0.8, lint.FailureCategoryComments,
-			`comment on exported %s %s should be of the form "%s..."`, kind, name, prefix,
+			`comment on exported %s %s should be of the form "%s..."`, kind, name, expectedPrefix,
+		)
+		return
+	}
+
+	// They have the same prefix, but the case is different.
+	// For example, the comment starts with "SendJson" but should start with "SendJSON".
+
+	currentPrefix := strings.Split(firstCommentLine, " ")[0]
+	if strings.HasPrefix(w.StripFirstRune(firstCommentLine), w.StripFirstRune(expectedPrefix)) {
+		// Only the first character differs, such as "sendJSON" became "SendJSON".
+		// so we consider the scope has changed.
+
+		w.addFailuref(fn.Doc, 1, lint.FailureCategoryComments,
+			`comment on exported %s %s should be of the form "%s..." to match its exported status, not "%s ..."`,
+			kind, name, expectedPrefix, currentPrefix,
 		)
+		return
 	}
+
+	w.addFailuref(fn.Doc, 1, lint.FailureCategoryComments,
+		`comment on exported %s %s should be of the form "%s..." by using its correct casing, not "%s ..."`,
+		kind, name, expectedPrefix, currentPrefix,
+	)
+}
+
+func (*lintExported) HasPrefixFold(s, prefix string) bool {
```

Comment:
The method does not actually fold strings; it's a case-insensitive version of `strings.HasPrefix`
Maybe 
```suggestion
func (*lintExported) HasPrefixInsensitive(s, prefix string) bool {
```
describes better what the method does.

Reply:
I inspired myself from strings.EqualFold, but I'm fine with your wording